src/components/VKButton.tsx

```diff
@@ -0,0 +1,19 @@
+import getVKCode from "../helpers/parseUrl";
+import { REDIRECT_URL, VK_APP, BACKEND_URL } from "../constants";
+
+const VKButton = () => {
+  const handleRedirect = () => {
+    window.location.href = `https://oauth.vk.com/authorize?client_id=${VK_APP.id}&display=popup&redirect_uri=${REDIRECT_URL}&scope=email&response_type=code&v=5.120&state=4194308`;
+    const code = getVKCode(window.location.href)!;
```

Comment:
please, use react router tools here to pick up current location, e.g. useLocation.

What does '!' at the line end mean here?

Reply:
Primarily, to resolve type conflict. But now it's unnecessary, removed.